scalr/data_source_current_run.go

```diff
@@ -132,30 +132,25 @@ func dataSourceScalrCurrentRunRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("environment_id", workspace.Organization.Name)
 
 	if workspace.VCSRepo != nil {
-		log.Printf("[DEBUG] Read ingress attributes of run: %s", runID)
-		ingressAttributes, err := scalrClient.ConfigurationVersions.ReadIngressAttributes(ctx, run.ConfigurationVersion.ID)
-		if err != nil {
-			if err == scalr.ErrResourceNotFound {
-				return fmt.Errorf("Could not find configuration version %s", run.ConfigurationVersion.ID)
-			}
-			return fmt.Errorf("Error retrieving ingress attributes: %v", err)
-		}
-
-		var commitConfig []map[string]interface{}
-		commit := map[string]interface{}{
-			"sha":     ingressAttributes.CommitSha,
-			"message": ingressAttributes.CommitMessage,
-			"author": map[string]interface{}{
-				"username": ingressAttributes.SenderUsername,
-			},
-		}
-
+		log.Printf("[DEBUG] Read vcs revision attributes of run: %s", runID)
 		var vcsConfig []map[string]interface{}
 		vcs := map[string]interface{}{
 			"repository_id": workspace.VCSRepo.Identifier,
 			"branch":        workspace.VCSRepo.Branch,
-			"commit":        append(commitConfig, commit),
 		}
+
+		if run.VcsRevision != nil {
+			var commitConfig []map[string]interface{}
+			commit := map[string]interface{}{
+				"sha":     run.VcsRevision.CommitSha,
+				"message": run.VcsRevision.CommitMessage,
+				"author": map[string]interface{}{
+					"username": run.VcsRevision.SenderUsername,
+				},
+			}
+			vcs["commit"] = append(commitConfig, commit)
```

Comment:
You can also do `"commit": []map[string]interface{}` in vcs and append to it inside the if statement. This approach would be more consistent with the rest of the provider.

Reply:
fixed